src/azstoragetorch/_client.py

```diff
@@ -185,17 +219,32 @@ def _download_with_retries(self, pos: int, length: int) -> bytes:
 
     def _get_download_stream(self, pos: int, length: int) -> Iterator[bytes]:
         try:
-            return self._generated_sdk_storage_client.blob.download(
+            response = self._generated_sdk_storage_client.blob.download(
                 range=f"bytes={pos}-{pos + length - 1}",
-                modified_access_conditions=azure.storage.blob._generated.models.ModifiedAccessConditions(
+            ) 
+            if self._blob_size is None: 
+                headers = response.response.headers
+                if "Content-Range" not in headers:
```

Comment:
It might be worth isolating this logic into a separate helper method `_set_blob_properties_from_download()` to reduce the length of the `_get_download_stream()` method.